src/main/java/pages/InventoryPage.java

```diff
@@ -31,28 +32,34 @@ public HeaderMenuComponent getHeaderMenuComponent() {
     }
 
     @Override
-    public boolean isPageOpened() {
-        return !inventoryItems.isEmpty() && inventoryItems.get(0).isElementPresent();
+    protected ExtendedWebElement getUniqueElement() {
+        return sortingDropdown;
     }
 
     public void addProductToCartByName(String productName) {
-        for (InventoryItemComponent item : inventoryItems) {
-            if (item.getProductName().equalsIgnoreCase(productName)) {
-                item.addToCart();
-                return;
-            }
-        }
-        throw new RuntimeException("Product not found in inventory: " + productName);
+        inventoryItems.stream()
+                .filter(item -> item.getProductName().equalsIgnoreCase(productName))
+                .findFirst()
+                .ifPresentOrElse(
+                        InventoryItemComponent::addToCart,
+                        () -> {
+                            throw new RuntimeException("Product not found in inventory: " + productName);
+                        }
+                );
     }
 
     public ProductPage openProductPageByProductName(String productName) {
-        for (InventoryItemComponent item : inventoryItems) {
-            if (item.getProductName().equalsIgnoreCase(productName)) {
-                item.getProductLink().click();
-                return new ProductPage(getDriver());
-            }
-        }
-        throw new RuntimeException("Product not found: " + productName);
+        inventoryItems.stream()
+                .filter(item -> item.getProductName().equalsIgnoreCase(productName))
+                .findFirst()
+                .ifPresentOrElse(
+                        item -> item.getProductLink().click(),
```

Comment:
Try to use method clickOnProductName in InventoryItemComponent instead of getProductLink